src/fastjet/_generalevent.py

```diff
@@ -627,6 +627,35 @@ def constituents(self, min_pt):
         res = ak.Array(self._replace_multi())
         return res
 
+    def exclusive_jets_constituents(self, njets):
+        if njets <= 0:
+            raise ValueError("Njets cannot be <= 0")
+
+        self._out = []
+        self._input_flag = 0
+        for i in range(len(self._clusterable_level)):
+            np_results = self._results[i].to_numpy_exclusive_njet_with_constituents(
+                njets
+            )
+            off = np.insert(np_results[-1], 0, 0)
```

Comment:
This numpy operation resembles the code that was causing problems with `inclusive_jets()`. The code was working fine but when tested at scale it was giving seg faults sporadically. In the end, we realized that the offset array needs to be `len+1` in the C++ code and here we just get the array: `off = np_results[-1]`. Now, I realize that I didn't remove all those parts :/ I am fine with keeping this code as is but keep in mind this problem. And we might want to remove all those lines in the future (including the code outside of this PR).

Reply:
I think it would be best to sanitize memory access before we merge this. @chrispap95 can you summarize a list of checks / fixes that should be addressed before we merge this?

Reply:
The fix is to change the allocation size for the offsets array in C++ to `{len+1}` from `{len}` and simplify python code from `off = np.insert(np_results[-1], 0, 0)` to `off = np_results[-1]`. Also, it would be fantastic if this can be done as well for any methods that I forgot to patch. @rkansal47 let me know if you have trouble doing this.

Reply:
OK, I believe I've fixed all of them (needed to just insert the 0 in C++ as well).